Add option to view coordinates on every square

gmanjon opened the pull request:
This way every square of the board has its coordinate in it with an overlay with a tiny opacity, so it doesn't bother too much, but enaugh to get used to them and learn them by playing, not boringly memorizing them.

gmanjon commented:
This is how it is displayed, the option menu and the overlay coordinates.

![Screenshot from 2020-06-17 01-43-21](https://user-images.githubusercontent.com/8287666/84838960-1bd9b400-b03c-11ea-87a2-43f99772b0c4.png)
![Screenshot from 2020-06-17 01-43-46](https://user-images.githubusercontent.com/8287666/84838966-1ed4a480-b03c-11ea-8a15-f9426128d6b9.png)

papb commented:
This is a fantastic idea!! Thank you, I will be definitely using this option.

veloce commented:
Hi! Thanks for the PR. I don't know how you translated all the strings but I won't be able to accept those changes. The translation files are generated by a script that pull data from a centralized translation service (crowdin), and this is the only way to modify them.

gmanjon commented:
Hi! Ok, didn't know that. It makes a lot of sense. I will update the PR removing all translations, but _**en-US.js**_ I guess, right?

Thank you!

veloce commented:
No, you don't touch any translation file, just use a simple string instead. A translation could be later added to crowdin for this but there are already several unstranlated string specific to the app.

gmanjon commented:
Hi veloce,

I just updated the PR removing all translations. Also, as the opacity setting usefullness depnds on the selected board theme I added settings to adjust opacity for white and black squares independently, with a sample board to live view the setting. I upload a screen recorder to show how it works. 

Please, tell me what you trully think. Any comments, critics or whatever are really welcome.


![ezgif-6-61f0066a6e7d](https://user-images.githubusercontent.com/8287666/85277058-1092e800-b483-11ea-8ebc-912285918f4c.gif)

papb commented:
Hi @gmanjon I think this is fantastic. May I ask how you generated that gif?

gmanjon commented:
Hi @papb, thanks! It's just the screen recorder tool of my mobile

veloce commented:
@gmanjon thanks for contributing to the project.

So to be honest I've a mixed feeling about this. 

On one hand this is an open source project and I want to encourage contributions. Your feature is well coded as far as I've seen. The only problem that I see are settings that are a bit too complex in my opinion, I think the transparency selector is too much.
On the other hand, even if the feature is small, it adds its cost of maintenance. I must consider this for every feature, however small it is. I must think of the ratio "usefulness/maintenance cost", and @ornicar does it too when adding new features to the website. And when coding this app I mostly follow the site features, trying to adapt them as much as possible for mobile usage.

So your feature will be useful for you and some part of the app users probably, but I'm not sure a lot of users will use it. 

Another reason, and the most important one, for what I'm a bit reluctant to integrate it, is that the lichess website has already a [coordinates training mode](https://lichess.org/training/coordinate). Since this is the official lichess app, the aim is to have the same features as the website eventually.
I feel that this training mode would be a fantastic addition to the app since it matches well mobile usage. It could even work offline. I'm not here to judge what is the best way to learn coordinates, of course. But I feel this training mode is a good way to learn coordinates, especially if you do a bit every day.

Perhaps you see me coming now :) What would be awesome if you want to contribute to the app, and are interested to do it, would be a PR to add this coordinate training mode in the app.

gmanjon commented:
Hi  @veloce 

I completely agree with everything. This is something I've been looking for in so many chess applications and couldn't find anywhere. Here I saw the oportunity to add it myself, as fortunately I am a software developer. But I understand that this is something that I (and may be more people too) would like, but I don't know the acceptance it would have on the whole community.

I also consider this addition a bit awkard in the sense that you mention, that this is the official app of a website, and so it should "go with it". So the first step would be to open a feature proposal (and see if people voted it) to add it to lichess.org in the first place. If we start integrating all the features people develop for themselves this will end up a complete mess.

So, no hurt feelings. I completely understand the reasons to not integrate it. I just had to develop it, I couldn't help it. I can use it locally either way :D

Regarding developing the current coordinates training mode I will look into in, but I'm afraid I cannot asure anything. Two little daughters are really time consuming... may be I'll teach the mithril hahahaha

Kind reagrds veloce, and thank you for all the work put in this app. It's really heroic.

papb commented:
Hi @veloce, I would really love to use this feature. Let's see if I can change your feelings.

> On one hand this is an open source project and I want to encourage contributions. Your feature is well coded as far as I've seen. The only problem that I see are settings that are a bit too complex in my opinion, I think the transparency selector is too much.

I disagree, the transparency setting only appears when enabled, it has a very user-friendly visualizer to see the effect; I don't think it's complex. My reaction is "wow this is fantastic, it even allows fine-tuning", definitely not "hmm complex".

> On the other hand, even if the feature is small, it adds its cost of maintenance. I must consider this for every feature, however small it is. I must think of the ratio "usefulness/maintenance cost", and @ornicar does it too when adding new features to the website.

Now that I've seen this PR, I think this PR is so useful that I can barely imagine playing without this feature anymore :sweat_smile:

> And when coding this app I mostly follow the site features, trying to adapt them as much as possible for mobile usage.

This is a good point, perhaps the feature should be added to the website first? How can I help with that?

> So your feature will be useful for you and some part of the app users probably, but I'm not sure a lot of users will use it.

I agree; certainly only a minority of users will use this.

> Another reason, and the most important one, for what I'm a bit reluctant to integrate it, is that the lichess website has already a coordinates training mode. Since this is the official lichess app, the aim is to have the same features as the website eventually.

Sure, the mobile app should have the coordinates training mode feature as well, since it should ideally have everything also present in the mobile app, but the fact that the lichess website has a coordinates training mode does not prevent the proposal of another feature to allow training coordinates in another way.

papb commented:
Let's not waste the work done by @gmanjon, I think this feature is so good, I'm in love with it already.

If in the future this ever becomes a maintenance burden, then you can remove it then. If we were discussing whether or not to implement the feature, I can see all your points, but since it's done already, let's just have it in action, please :grimacing: :grimacing:

gmanjon commented:
Hi @papb. Thank you for the compliments :D

@veloce, I just want to say that, in case this feature need maintenance due to bug reports, needed tweaks or evolutives I compromise to take responsability of all those developments.

I don't want to push it, I already told I understand all those reasons, truthfully. I just wanted to be clear.

Kind regards to everyone

Ka-hu commented:
:+1:  This is a good feature, useful for beginners. Lichess.org could also integrate this feature.

LichessUser commented:
@veloce to be honest I love this feature and it would be really helpful for new players to learn the language of chess. With this feature, it'll become easy for us to write our games.
@gmanjon has done a lot of hardwork for this, let's not waste it and integrate it to the app.
Sorry for my had English.
Thank you!

LichessUser commented:
@gmanjon can you please work on arrows and circles on the lichess app?
I am pretty sure that many players will use it as it is really a cool feature for the mobile players.
Btw, thank you for this coordinates on every square feature, I am not sure if it really get added but the work you done is just fantastic.
Greetings from India

veloce commented:
@LichessUser you already created an issue about it, and that's enough. Now please stop spamming people.

LichessUser commented:
@Veloce Ok I won't do it next time, sorry about that.

gmanjon commented:
Hi everyone,

Thanks for the compliments @LichesUser. But I must say that it wasn't a great effort at all. Even more, I completely enjoyed it. I don't want anyone to feel any preassure due to the work put on this task.

I even don't think that a big effort (which it wasn't) should be a reason to merge any feature. All features should be discussed, ideally before any line of code is typed. I just didn't follow the standard procedure just to save time to @veloce, but I now realize that this was a mistake, I should have started the discussion before making any PR. Now people are "requiring" instead of "discussing" adn trying to prove some of the argument. People sould be asked, make a pol or whatever. 

I completely agree with @veloce that this feature should not be merged unless someone can prove that this feature likes to an enough amount of users. And even then there is this matter that mobile app should follow website, which doesn't have this feature.

Kind regards

aldokkani commented:
Fantastic work, would love to see it on the app.
